### PR TITLE
Update the declaration of struct for FFI downcall tests

### PR DESCRIPTION
**Pull request by ChengJin01**
Fixes: #15134

Signed-off-by: Cheng Jin <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
Reviewer: @keithc-ca
FYI: @tajila, @pshipton

File: runtime/tests/clinkerffi/downcall.h [hunk 1]
```diff
@@ -34,450 +34,370 @@
 
 typedef int64_t LONG;
```
Comment (keithc-ca) [inline runtime/tests/clinkerffi/downcall.h:35]: This file should be bracketed by
```
#if !defined(downcall_h)
...
#endif /* !defined(downcall_h) */
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/tests/clinkerffi/downcall.h (hunk 1 above)
Comment (ChengJin01) [reply runtime/tests/clinkerffi/downcall.h:37]: Added as suggested above.

File: runtime/tests/clinkerffi/downcall.h [hunk 2]
```diff
@@ -27,457 +27,381 @@
  * Created by jincheng@ca.ibm.com
  */
 
+#if !defined(downcall_h)
+
```
Comment (keithc-ca) [inline runtime/tests/clinkerffi/downcall.h:31]: Of course, this needs `#define downcall_h` for the `#if` to be effective.

Comment (ChengJin01) [reply runtime/tests/clinkerffi/downcall.h:32]: Follow the existing coding convention for header to replace with:
```
#ifndef DOWNCALL_H
#define DOWNCALL_H

#endif /* DOWNCALL_H */
```

**Comment by keithc-ca** [comment]
jenkins compile alinux64 jdk17

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**